src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -457,16 +459,44 @@ final class RiskProvider: RiskProviding {
 	
 	private func checkIfRiskLevelHasChangedForNotifications(_ risk: Risk) {
 		/// Triggers a notification for every risk level change.
-		if risk.riskLevelHasChanged {
-			Log.info("Trigger notification about changed risk level", log: .riskDetection)
-			UNUserNotificationCenter.current().presentNotification(
-				title: AppStrings.LocalNotifications.detectExposureTitle,
-				body: AppStrings.LocalNotifications.detectExposureBody,
-				identifier: ActionableNotificationIdentifier.riskDetection.identifier
-			)
+		switch risk.riskLevelChange {
+		case .decreased:
+			Log.info("decrease risk change state won't trigger a high risk notification")
+		case .increased:
+			triggerHighRiskNotification()
+		case let .unchanged(riskLevel):
+			guard riskLevel == .high,
+				  let mostRecentDateWithRiskLevel = risk.details.mostRecentDateWithRiskLevel,
+				  let previousMostRecentDateWithRiskLevel = store.mostRecentDateWithRiskLevel,
+				  mostRecentDateWithRiskLevel > previousMostRecentDateWithRiskLevel
+			else {
+					  Log.info("Missing mostRecentDateWithRiskLevel - do not trigger anything")
+					  return
+				  }
+			triggerHighRiskNotification()
```

Comment:
If the app is processing this in the background, a notification will trigger here.
And then if the user starts the app, another notification will show up because of `store.showAnotherHighExposureAlert = true`. Is this correct?